### PR TITLE
DOCS-#3322: Specify proper OS versions for which installation packages are available

### LINKED ISSUE #3322: Specify proper OS versions for which installation packages are available
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
- **Modin version** (`modin.__version__`):
- **Python version**:
- **Code we can use to reproduce**:

### Describe the problem

We need to clarify in docs for what OS each installation channel are available, eg conda's modin-omnisci is available only on Linux at the moment

### PR DESCRIPTION
**Pull request by Garra1980**
Signed-off-by: izamyati <igor.zamyatin@intel.com>

## What do these changes do?
Update info about OSs where modin is available in documentation

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #3322 
- [ ] tests added and passing
- [x] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

File: docs/installation.rst [hunk 1]
```diff
@@ -12,7 +12,7 @@ Installing with pip
 Stable version
 """"""""""""""
 
-Modin can be installed with pip. To install the most recent stable release run the following:
+Modin can be installed with pip both on Linux and Windows. To install the most recent stable release run the following:
```
Comment (vnlitvinov) [inline docs/installation.rst:15]: Modin also supports MacOS

File: docs/installation.rst [hunk 2]
```diff
@@ -62,19 +62,19 @@ Using conda-forge channel
...
-| modin-ray                       | Ray                       |
-+---------------------------------+---------------------------+
-| modin-omnisci                   | OmniSci                   |
-+---------------------------------+---------------------------+
++---------------------------------+---------------------------+----------------------+
+| **Package name in conda-forge** | **Engine(s)**             | **Supported OSs**    |
++---------------------------------+---------------------------+----------------------+
+| modin                           | Dask_                     |   Linux, Windows     |
++---------------------------------+---------------------------+----------------------+
+| modin-dask                      | Dask                      |   Linux, Windows     |
```
Comment (vnlitvinov) [inline docs/installation.rst:70]: Dask supports all three OSes including Windows

File: docs/installation.rst [hunk 3]
```diff
@@ -12,7 +12,8 @@ Installing with pip
 Stable version
 """"""""""""""
 
-Modin can be installed with pip both on Linux and Windows. To install the most recent stable release run the following:
+Modin can be installed with pip both on Linux and Windows. Also MacOS is supported for Dask_ engine (see below).
```
Comment (vnlitvinov) [inline docs/installation.rst:15]: This is still wrong :) there is Ray for MacOS on `pypi`, so Modin with Ray can be installed on MacOS via `pip` but cannot via `conda`.

Comment (Garra1980) [reply docs/installation.rst]: Crap! Fixed

**Review by vnlitvinov — APPROVED** [review]

**Pull request merged**